app/scripts/views/settings/clients.js

```diff
@@ -115,41 +136,52 @@ define(function (require, exports, module) {
       });
     },
 
-    _onDeviceAdded: function () {
+    _onItemAdded: function () {
       this.render();
     },
 
-    _onDeviceRemoved: function (device) {
-      var id = device.get('id');
-      var self = this;
-      $('#' + id).slideUp(DEVICE_REMOVED_ANIMATION_MS, function () {
-        // re-render in case the last device is removed and the
-        // "no registered devices" message needs to be shown.
-        self.render();
-      });
+    _onItemRemoved: function (item) {
+      $('#' + item.get('id')).slideUp(DEVICE_REMOVED_ANIMATION_MS);
     },
 
-    _onDisconnectDevice: function (event) {
-      this.logViewEvent('disconnect');
-      var deviceId = $(event.currentTarget).attr('data-id');
-      this._destroyDevice(deviceId);
+    _onDisconnectClient: function (event) {
+      var itemId = $(event.currentTarget).data('id');
```

Comment:
Can you make this two separate functions, the first to get the itemId and clientType from the event, the second to do the other work?